bigtable-hbase-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java

```diff
@@ -63,51 +78,148 @@
   private final BigtableOptions options;
   private final BigtableTableAdminClient bigtableTableAdminClient;
   private BigtableInstanceName bigtableInstanceName;
+  private final TableAdapter2x tableAdapter2x;
 
   public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOException {
     LOG.debug("Creating BigtableAsyncAdmin");
     this.options = asyncConnection.getOptions();
     this.bigtableTableAdminClient = asyncConnection.getSession().getTableAdminClient();
     this.disabledTables = asyncConnection.getDisabledTables();
     this.bigtableInstanceName = options.getInstanceName();
+    this.tableAdapter2x = new TableAdapter2x(options, new ColumnDescriptorAdapter());
   }
 
   @Override
   public CompletableFuture<Void> createTable(TableDescriptor desc, Optional<byte[][]> splitKeys) {
-    throw new UnsupportedOperationException("createTable"); // TODO
+    if (desc.getTableName() == null) {
+      return FutureUtils.failedFuture(new IllegalArgumentException("TableName cannot be null"));
+    }
+
+    return CompletableFuture.supplyAsync(() -> {
+      CreateTableRequest.Builder builder = CreateTableRequest.newBuilder();
+      builder.setParent(bigtableInstanceName.toString());
+      builder.setTableId(desc.getTableName().getQualifierAsString());
+      builder.setTable(tableAdapter2x.adapt(desc));
+      if (splitKeys.isPresent()) {
+        for (byte[] splitKey : splitKeys.get()) {
+          builder
+              .addInitialSplits(Split.newBuilder().setKey(ByteString.copyFrom(splitKey)).build());
+        }
+      }
+      return builder;
+    }).thenCompose(
+        r -> FutureUtils.toCompletableFuture(bigtableTableAdminClient.createTableAsync(r.build())))
+        .thenAccept(r -> {
+        });
   }
 
   @Override
   public CompletableFuture<Void> createTable(TableDescriptor desc, byte[] startKey, byte[] endKey,
       int numRegions) {
-    throw new UnsupportedOperationException("createTable"); // TODO
+
+    return CompletableFuture.supplyAsync(() -> {
```

Comment:
why not build the request, and then just call `return createTable(desc, skeys)`?  Why do you need a CompletableFuture?